src/carousel/carousel.ts

```diff
@@ -60,11 +68,11 @@ export class NgbSlide {
   },
   template: `
     <ol class="carousel-indicators" *ngIf="showNavigationIndicators">
-      <li *ngFor="let slide of slides" [id]="slide.id" [class.active]="slide.id === activeId"
+      <li *ngFor="let slide of slides" [id]="'ngb-slide-indicator-' + slide.id" [class.active]="slide.id === activeId"
           (click)="select(slide.id, NgbSlideEventSource.INDICATOR)"></li>
     </ol>
     <div class="carousel-inner">
-      <div *ngFor="let slide of slides" class="carousel-item" [class.active]="slide.id === activeId">
+      <div *ngFor="let slide of slides" class="carousel-item" [id]="'ngb-slide-' + slide.id">
```

Comment:
This generates the following ids: ex. `ngb-slide-ngb-slide-13`, why is the `ngb-slide-` prefix necessary?
Can't we just use `slide.id`?

Reply:
If I'm not wrong, having dom ids starting with numbers is a problem. It's just to avoid this case.

Reply:
it is already prefixed with `ngb-slide-`, you're prefixing it twice

Reply:
Yes, you're right, I answered too quickly 😟. I think I prefixed it the same way than the the indicators, but it's not needed. I'll remove it.

Reply:
Finally, I need to prefix it. We prefixed the id only when it's not provided, this way:

```
@Input() id = `ngb-slide-${nextId++}`;
```

See the test `should mark the requested slide as active` for example.

Reply:
What about not prefixing it when provided, but making sure they **do not** provide a number as ID ?

Reply:
IMO, I think the simpler (what have done in the latest version) is to prefix it in the template, not in the generated id. It keeps the freedom to provide what you want, without having to check anything.